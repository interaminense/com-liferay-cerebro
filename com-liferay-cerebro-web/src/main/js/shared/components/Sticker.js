import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';
import {isString} from 'lodash';

import Icon from './Icon';

const DISPLAYS = [
	'primary',
	'secondary',
	'success',
	'info',
	'warning',
	'danger',
	'light',
	'dark'
];

const SIZES = ['sm', 'lg'];

class Sticker extends Component {
	render() {
		const {display, size, symbol} = this.props;

		const classes = getCN(
			'sticker',
			'sticker-root',
			'sticker-rounded',
			'sticker-static',
			{
				[`sticker-${display}`]: display,
				[`sticker-${size}`]: size
			}
		);

		return (
			<div {...this.otherProps()} class={classes}>
				<Icon symbol={symbol} />
			</div>
		);
	}
}

Sticker.PROPS = {
	display: Config.oneOf(DISPLAYS),
	elementClasses: Config.string(),
	size: Config.oneOf(SIZES),
	symbol: Config.string().required()
};

export function hashCode(s) {
	return s.split('').reduce((a, b) => a + b.charCodeAt(0), 0);
}

export function getDisplayForId(id) {
	if (isString(id)) {
		id = hashCode(id);
	}

	return DISPLAYS[id % DISPLAYS.length];
}

Sticker.DISPLAYS = DISPLAYS;
Sticker.SIZES = SIZES;

export default Sticker;