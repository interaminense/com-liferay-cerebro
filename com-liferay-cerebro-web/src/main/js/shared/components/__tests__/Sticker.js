import Sticker, {getDisplayForId, hashCode} from '../Sticker';

describe('Sticker', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new Sticker({
			symbol: 'file'
		});

		expect(component).toMatchSnapshot();
	});
});

describe('getDisplayForId', () => {
	it('should return a display type for a given number or string', () => {
		expect(getDisplayForId(1234)).toBe('success');
		expect(getDisplayForId('foo')).toBe('warning');
	});
});

describe('hashCode', () => {
	it('should return a unique number for a given string', () => {
		expect(hashCode('foo')).toBe(324);
		expect(hashCode('bar')).toBe(309);
		expect(hashCode('baz')).toBe(317);
	});
});