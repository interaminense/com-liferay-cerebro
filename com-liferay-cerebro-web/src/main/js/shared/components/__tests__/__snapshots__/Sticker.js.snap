// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sticker should render 1`] = `
<div class="sticker sticker-root sticker-rounded sticker-static">
  <svg class="icon-root lexicon-icon lexicon-icon-file">
    <use xlink:href="/images/lexicon/icons.svg#file"></use>
  </svg>
</div>
`;
